## Hello Everyone!

Welcome backk! This time i will show you how to do data preprocessing. Before we discuss more further about data preprocessing, first we must know what's the definition and the purpose of them. 

- Data Preprocessing: Data Preprocessing is one of several steps in data mining. The purpose of this step is to make the data more efficient for mining process.

So are you ready for do data preprocessing? So let's do it together!

First step we must import the data into the notebook (python). Do you still remember how to import dataset into the notebook? Good!. If you don't remember, you can check the previous tutorial on my github or in this link: https://github.com/mannaufal/Load-Dataset-In-Python.

first we load pandas library
pandas is library that can be used for importing dataset to the notebook. pd is an initial for the library

In [1]:
import pandas as pd

After we load the library, now we can import the dataset.

In [2]:
df = pd.read_csv('iris.csv')

Good!, the data has been imported to the notebook. To make sure the data was imported, we must show it in the notebook.

In [3]:
df

sepal length  sepal width  petal length  petal width           class
0             5.1          3.5           1.4          0.2     Iris-setosa
1             4.9          3.0           1.4          0.2     Iris-setosa
2             4.7          3.2           1.3          0.2     Iris-setosa
3             4.6          3.1           1.5          0.2     Iris-setosa
4             5.0          3.6           1.4          0.2     Iris-setosa
..            ...          ...           ...          ...             ...
145           6.7          3.0           5.2          2.3  Iris-virginica
146           6.3          2.5           5.0          1.9  Iris-virginica
147           6.5          3.0           5.2          2.0  Iris-virginica
148           6.2          3.4           5.4          2.3  Iris-virginica
149           5.9          3.0           5.1          1.8  Iris-virginica

[150 rows x 5 columns]

### HANDLING MISSING VALUES

Great! The dataset has been imported to the notebook. Now, we can check if there's a missing value or not in the dataset. We can type this script "df.isna().sum()" in the kernel below.

In [4]:
df.isna().sum()

sepal length    0
sepal width     0
petal length    0
petal width     0
class           0
dtype: int64

### DUPLICATES DATA CHECK

The next step, we must make sure the data doesn't has any duplicate data

In [5]:
df[df.duplicated()]

sepal length  sepal width  petal length  petal width           class
34            4.9          3.1           1.5          0.1     Iris-setosa
37            4.9          3.1           1.5          0.1     Iris-setosa
142           5.8          2.7           5.1          1.9  Iris-virginica

as we can see, in rows 34, 37, and 142 there are 3 duplicate datas. so we must remove the duplicate data.

In [6]:
df.drop_duplicates(keep=False,inplace=True)

once again, we must make sure the data is really doesn't has any duplicate data.

In [8]:
df[df.duplicated()]

Empty DataFrame
Columns: [sepal length, sepal width, petal length, petal width, class]
Index: []

Great work! You have learned how to data preprocessing using python. I only show the simple from data preprocessing, there are many step for data preprocessing like normalization, data reduction, data integration, and etc. It's pretty easy right huh?. Thank you for your attention. I hope this tutorial can help you. See ya!